backend/src/domain/error.ts

```diff
@@ -0,0 +1,20 @@
+export const DatabaseMigrationError = 'DATABASE_MIGRATION is optional, but if set must be one of [UP, DOWN]';
+export const DatabaseConnectionError = 'Missing environment variables for database connection';
+export const EntryNotFoundError = (tableName: string, filter: any) =>
+  `Unable to find entry in ${tableName} with filter [${JSON.stringify(filter)}]`;
+export const InsertError = (resultLength: number) => `Insert expected one element but was ${resultLength}`;
+export const MealNotFoundError = (mealName: string) => `Couldn't find meal with name:  ${mealName.toUpperCase()})`;
+export const Level60Error = 'Somehow level 60 produced ingredients not populated for buddy-combination-dao';
+export const EnergyPercentageError = (energyPercentage: number) =>
+  `Energy has to be between 0 and 150, but was ${energyPercentage}`;
+export const LevelError = 'Only level 1-100 allowed';
+export const NatureError = (natureName: string) => `Couldn't find nature with name:  ${natureName.toUpperCase()}`;
+export const PokemonNameError = (pokemonName: string) =>
+  `Couldn't find nature with name:  ${pokemonName.toUpperCase()}`;
+export const SubTeamsError = 'Should not happen since we check maxTeamSize > 1, and we return only undefined for 0';
+export const IngredientError = (name: string) => `Ingredient with name [${name}] does not exist`;
+export const ProgrammerError = (name: string) =>

```

Comment:
I would like to rename `ProgrammerError` to `ProgrammingError` as it suggest an issue with the code rather than blaming an individual programmer. This can be moved into `domain/error/programming/programming-error.ts`